src/Inputs/scene.inputManager.ts

```diff
@@ -804,64 +806,81 @@ export class InputManager {
             }
         };
 
-        let attachedFunction = () => {
-            if (!elementToAttachTo || this._keyboardIsAttached) {
-                return;
-            }
-            elementToAttachTo.addEventListener("keydown", this._onKeyDown, false);
-            elementToAttachTo.addEventListener("keyup", this._onKeyUp, false);
-            this._keyboardIsAttached = true;
-        };
+        this._deviceSourceManager.onDeviceConnectedObservable.add((device) => {
+            // Keyboard Events
+            if (device.deviceType == DeviceType.Keyboard) {
+                this._deviceSourceManager.getDeviceSource(device.deviceType, device.deviceSlot)!.onInputChangedObservable.add((deviceSource) => {
+                    if (elementToAttachTo && document.activeElement === elementToAttachTo) {

```

Comment:
I would recommand early exit to clean the code:
```
if (!elementToAttachTo || document.activeElement !== elementToAttachTo) {
return;
}
```

Reply:
That being said I like that approach, it is better than the focus/blur way we did before

Reply:
Changed